### PR TITLE
Ensure NbserverListApp produces valid json even for multiple/no servers

### PR DESCRIPTION
**Pull request by jcb91**
This PR makes a small adjustment to the NbserverListApp class to ensure valid json output.

Currently, the NbserverListApp outputs a json object for each running server, without wrapping them in a list. This means that if there are no running servers, there is no output, and if multiple servers are running, the response is invalid json.

### CODE REVIEW

File: notebook/notebookapp.py [hunk 1]
```diff
@@ -398,12 +398,12 @@ class NbserverListApp(JupyterApp):
...
     def start(self):
-        if not self.json:
-            print(_("Currently running servers:"))
-        for serverinfo in list_running_servers(self.runtime_dir):
-            if self.json:
-                print(json.dumps(serverinfo))
-            else:
+        serverinfo_list = list(list_running_servers(self.runtime_dir))
+        if self.json:
+            print(json.dumps(serverinfo_list))
```
Comment (takluyver) [inline notebook/notebookapp.py]: Maybe pass `indent=2` so that the JSON is nicely formatted if a human needs to look at it?

**Comment by takluyver** [comment]
Correction: it's been there since IPython 2.0, released in April 2014. See ipython/ipython#4772. We discussed making it single JSON array then, but I was lazy (and apparently just dumping the whole list as JSON didn't occur to me :confused: ).

I think we should make this behaviour a new option, and then maybe go through a deprecation cycle with `--json`. Even if we think no-one is using the old behaviour, we can't be sure, and I think we should take backwards compatibility a bit more seriously than we do.

**Comment by jcb91** [comment]
Sounds reasonable to me, I'll update this to add a separate option when I get a chance

File: notebook/notebookapp.py (hunk 1 above)
Comment (jcb91) [inline notebook/notebookapp.py]: :+1:

**Comment by takluyver** [comment]
Thanks for being so understanding about it! :-)

**Comment by jcb91** [comment]
ok, I've added a (new) separate option `jsonlist`. For now, this is used in preference to the plain `json` flag if both are specified; is that ok, or would you rather they were made mutually exclusive, like the `--sys`, `--user` and `--sys-prefix` flags of `jupyter nbextension` and `jupyter serverextension`?

**Comment by takluyver** [comment]
Thanks, that looks good to me. I'm not worried about the behaviour when both are specified.

I'll let @gnestor have a look - if you're happy, feel free to merge.

**Pull request merged**